### PR TITLE
Try to predict icon pack package based on export path

### PR DESCRIPTION
**Pull request by egorikftp**
https://github.com/user-attachments/assets/3c391f05-66eb-45bd-bb57-86d61078a625

### CODE REVIEW

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationScreen.kt [hunk 1]
```diff
@@ -118,20 +118,39 @@ private fun IconPackDestinationScreenUI(
...
+                            text = "Predicted package:",
+                            style = MaterialTheme.typography.bodyMedium,
+                        )
+                        Text(
+                            modifier = Modifier.align(Alignment.Start),
+                            textDecoration = TextDecoration.Underline,
+                            text = state.predictedPackage,
+                            style = MaterialTheme.typography.bodySmall,
+                        )
+                    }
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationScreen.kt:153]:
```suggestion
                    Text(
                        modifier = Modifier.align(Alignment.Start),
                        text = "Export path:",
                        style = MaterialTheme.typography.bodyMedium,
                    )
                    Text(
                        modifier = Modifier.align(Alignment.Start),
                        text = state.iconPackDestination,
                        textDecoration = TextDecoration.Underline,
                        style = MaterialTheme.typography.bodySmall,
                    )
                }
                if (state.predictedPackage.isNotEmpty()) {
                    Text(
                        modifier = Modifier.align(Alignment.Start),
                        text = "Predicted package:",
                        style = MaterialTheme.typography.bodyMedium,
                    )
                    Text(
                        modifier = Modifier.align(Alignment.Start),
                        textDecoration = TextDecoration.Underline,
                        text = state.predictedPackage,
                        style = MaterialTheme.typography.bodySmall,
                    )
```

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt [hunk 2]
```diff
@@ -27,6 +29,7 @@ class IconPackDestinationViewModel(
             copy(
                 iconPackDestination = path.absolutePathString(),
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt:30]:
```suggestion
                iconPackDestination = path.invariantSeparatorsPathString,
```

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt [hunk 3]
```diff
@@ -27,6 +29,7 @@ class IconPackDestinationViewModel(
...
                 nextButtonEnabled = true,
+                predictedPackage = PackageExtractor.getFrom(path.absolutePathString()).orEmpty(),
```
Comment (Goooler) [inline idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt:32]:
```suggestion
                predictedPackage = PackageExtractor.getFrom(path.invariantSeparatorsPathString).orEmpty(),
```

**Review by Goooler — APPROVED** [review]

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt (hunk 3 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt:32]: make sense now

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationScreen.kt (hunk 1 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationScreen.kt:153]: I wrapped blocks into `Column` to automatically handle paddings in root `Column` by `verticalArrangement = Arrangement.spacedBy(24.dp)`

File: idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt (hunk 3 above)
Comment (egorikftp) [reply idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/destination/IconPackDestinationViewModel.kt]: I decided to add `invariantSeparatorsPathString` logic only into `PackageExtractor`

File: components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        "src/jsMain/kotlin",
+        "src/iosMain/kotlin",
+        "src/wasmJsMain/kotlin",
+        "src/linuxMain/kotlin",
+        "src/macosMain/kotlin",
+        "src/nativeMain/kotlin",
+    )
+
+    fun getFrom(path: String): String? {
+        val invariantPath = path.toPath().invariantSeparatorsPathString
```
Comment (Goooler) [inline components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt:25]:
```suggestion
        val invariantPath = Path(path).invariantSeparatorsPathString
```

Comment (Goooler) [reply components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt:25]: But I would still suggest processing this step in the outsides.

File: components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            invariantPath.substringAfter(sourceSet)
+                .replace("/", ".")
+                .removePrefix(".")
+                .removeSuffix(".")
+        } else {
+            null
+        }
+    }
+
+    private fun String.toPath(): Path = Paths.get(this)
```
Comment (Goooler) [inline components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt:38]:
```suggestion
```

**Review by Goooler — APPROVED** [review]

File: components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt (hunk 4 above)
Comment (egorikftp) [reply components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt]: Could you please take a look?
https://github.com/ComposeGears/Valkyrie/compare/2caa814b2d8a5201411d5d6643aa09e881393b2e..ef5f451e30aee9158a01fe9a1d109a1a5882fc5e

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/String.kt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package io.github.composegears.valkyrie.extensions
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.invariantSeparatorsPathString
+
+fun String.toInvariantSeparatorsPathString() = toPath().invariantSeparatorsPathString
```
Comment (Goooler) [inline components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/String.kt:7]: Again, this is dirty, we should avoid converting string to path and handle path. If you really want this, please do it in `getFrom`.

Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/String.kt:7]: yes, it a reason why I decided to put this logic only into `PackageExtractor`. `iconPackDestination` from persistent settings are always String

Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/String.kt:7]: We have pure `Path` object only when user choose some destination path for icon pack

Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/String.kt:7]: > Again, this is dirty, we should avoid converting string to path and handle path. 

Don't see any alternative 🙂

**Pull request merged**